app/services/district_report_service.rb

```diff
@@ -0,0 +1,83 @@
+class DistrictReportService
+  def initialize(facilities:, selected_date:)
+    @facilities = Array(facilities)
+    @selected_date = selected_date
+    @data = {
+      controlled_patients: {},
+      registrations: {},
+      cumulative_registrations: 0,
+      quarterly_registrations: []
+    }.with_indifferent_access
+  end
+
+  attr_reader :selected_date, :facilities, :data
+
+  def call
+    compile_control_and_registration_data
+
+    compile_cohort_trend_data
+
+    data
+  end
+
+  def controlled_patients(time)
+    end_range = time.end_of_month
+    mid_range = time.advance(months: -1).end_of_month
+    beg_range = time.advance(months: -2).end_of_month
+    sub_query = LatestBloodPressuresPerPatientPerMonth
+      .select("distinct on (patient_id) *")
+      .under_control
+      .order("patient_id, bp_recorded_at DESC, bp_id")
+      .where(registration_facility_id: facilities)
+      .where("(year = ? AND month = ?) OR (year = ? AND month = ?) OR (year = ? AND month = ?)",
+        beg_range.year.to_s, beg_range.month.to_s,
+        mid_range.year.to_s, mid_range.month.to_s,
+        end_range.year.to_s, end_range.month.to_s)
+    LatestBloodPressuresPerPatientPerMonth.from(sub_query, "latest_blood_pressures_per_patient_per_months")
```

Comment:
Because we need to apply a `count` on the `distinct on` query, and those can't be combined into one query using ActiveRecord. 

Its the same pattern we use in the [blood pressure query](https://github.com/simpledotorg/simple-server/blob/faf5afd3232a3bc80e5a9a7829959e71031c8958/app/queries/my_facilities/blood_pressure_control_query.rb#L111-L132) for My Facilities.

Reply:
Ah this. 😂 😭 As you were...